kedro/io/partitioned_data_set.py

```diff
@@ -217,7 +220,7 @@ def _normalized_path(self) -> str:
     @cachedmethod(cache=operator.attrgetter("_partition_cache"))
     def _list_partitions(self) -> List[str]:
         return [
-            path
+            _grandparent(path) if self._dataset_config.get(VERSION_KEY) else path
```

Comment:
For readability maybe refactor it into a variable?
```python
dataset_versioned = self._dataset_config.get(VERSION_KEY, False)
return [
    _grandparent(path) if dataset_versioned else path
    for path in self._filesystem.find(self._normalized_path, **self._load_args)
    if path.endswith(self._filename_suffix)
]
```

Reply:
```suggestion
            _grandparent(path) if dataset_is_versioned else path
```